File: Java/String_2/extraEnd.java
/**
 * @author Luis Teixeira
 *
 * @date   04/04/2018
 *
 * @mail   filipe.teixeira.996@gmail.com
 */
package String_2;

import static org.junit.Assert.*;

import org.junit.Test;

/**

Given a string, return a new string made of 3 copies of the last 2 chars of the original string. The string length will be at least 2.


extraEnd("Hello") --> "lololo"
extraEnd("ab") --> "ababab"
extraEnd("Hi") --> "HiHiHi"
 */
public class extraEnd {

	private static int i = 1;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("extraEnd(\"Hello\") --> \"lololo\"\n");
		assertEquals("extraEnd(\"Hello\") --> \"lololo\"", "lololo", extraEnd("Hello"));
		i++;
	}

	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("extraEnd(\"ab\") --> \"ababab\"\n");
		assertEquals("extraEnd(\"ab\") --> \"ababab\"", "ababab", extraEnd("ab"));
		i++;
	}
	
	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("extraEnd(\"Hi\") --> \"HiHiHi\"\n");
		assertEquals("extraEnd(\"Hi\") --> \"HiHiHi\"", "HiHiHi", extraEnd("Hi"));
		i++;
	}
	
	public String extraEnd(String str) {

		String ret = "";
		
		if (str.length()==2) {
			ret = str+str+str;
			return ret;
		}
		
		else {
			ret = str.substring(str.length()-2, str.length());
			return ret+ret+ret;
		}
	}
}

File: Python/String-1/extra_end.py
import unittest

'''

Given a string, return a new string made of 3 copies of the last 2 chars of the original string.
The string length will be at least 2.


extra_end('Hello') -> 'lololo'
extra_end('ab') -> 'ababab'
extra_end('Hi') -> 'HiHiHi'

'''

def extra_end(str):
    
    char2 = str[len(str)-2]+str[len(str)-1]
 
    return 3*char2  


class Test_extra_end(unittest.TestCase):

    def test_1(self):
        self.assertEqual(extra_end('Hello'), 'lololo')

    def test_2(self):
        self.assertEqual(extra_end('Hello'), 'lololo')

    def test_3(self):
        self.assertEqual(extra_end('Hello'), 'lololo')


if __name__ == '__main__':
    unittest.main()   